mock-identity-system/src/main/java/io/mosip/esignet/mock/identitysystem/validator/IdentitySchemaValidator.java

```diff
@@ -0,0 +1,113 @@
+package io.mosip.esignet.mock.identitysystem.validator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.networknt.schema.*;
+import io.mosip.esignet.mock.identitysystem.dto.IdentityData;
+import io.mosip.esignet.mock.identitysystem.dto.RequestWrapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Set;
+
+@Component
+@Slf4j
+public class IdentitySchemaValidator implements ConstraintValidator<IdentitySchema, Object> {
+
+    @Value("${mosip.mock.ida.identity.schema.url}")
+    private String identitySchemaUrl;
+
+    @Value("#{${mosip.mock.ida.update-identity.non-mandatory.fields}}")
+    private Set<String> nonMandatoryFieldsOnUpdate;
+    private boolean isCreate;
+
+    private volatile JsonSchema schema;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    ResourceLoader resourceLoader;
+
+
+    @Override
+    public void initialize(IdentitySchema constraintAnnotation) {
+        this.isCreate = constraintAnnotation.isCreate();
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext context) {
+        context.disableDefaultConstraintViolation();
+        if (!(object instanceof RequestWrapper)) {
+            context.buildConstraintViolationWithTemplate("Invalid request type")
+                    .addConstraintViolation();
+            return false;
+        }
+        RequestWrapper wrapper= (RequestWrapper) object;
+        Object requestObject = wrapper.getRequest();
+        if (!(requestObject instanceof IdentityData)) {
+            context.disableDefaultConstraintViolation();
+            context.buildConstraintViolationWithTemplate("Invalid request object type").addConstraintViolation();
```

Comment:
We could directly return false from here, so that it will return "invalid_identity_data" error code. Or we can simply use our common "invalid_request" error code. The latter seems to be more appropriate. What do you think?

Reply:
ok no promlem we can do that